src/tests/functional/plugin/shared/src/snippets/convert.cpp

```diff
@@ -0,0 +1,186 @@
+    std::pair<std::vector<ov::element::Type>, std::vector<ov::element::Type>> types;
+    std::tie(inputShape, types, ref_num_nodes, ref_num_subgraphs, targetDevice) = this->GetParam();
+    init_input_shapes(static_shapes_to_test_representation(inputShape));
+
+    auto f = ov::test::snippets::ConvertFunction(inputShape, types.first[0], types.second[0]);
+    function = f.getOriginal();
+    output_type = types.second.front();
+}
+
+parameters Convert::generate_params_random() const {
+    int32_t startFrom, range, resolution = 5;
+    switch (output_type) {
+        case ov::element::f32:
+        case ov::element::i32:
+        case ov::element::bf16:
+            startFrom = -10;
+            range = 20;
+            break;
+        case ov::element::u8:
+            startFrom = -10;
+            range = 20;
+            break;
+        case ov::element::i8:
+            startFrom = 117;
+            range = 20;
+            break;
+        default:
+            startFrom = 0;
+            range = 10;
+    }
+    return {{ startFrom, range, resolution }};
+}
+
+void Convert::generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) {
+    inputs.clear();
+    const auto& funcInputs = function->inputs();
+    const auto params = generate_params_random();
+    if (params.size() != funcInputs.size()) {
+        IE_THROW() << "Incorrect count of parameters for random generation and inputs of function!";
+    }
+
+    for (int i = 0; i < funcInputs.size(); ++i) {
+        const auto& funcInput = funcInputs[i];
+        ov::Tensor tensor;
+        int32_t startFrom, range, resolution;
+        std::tie(startFrom, range, resolution) = params[i];
+        tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i],
+                                                         range, startFrom, resolution);
+        inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+    }
+}
+
+void ConvertInput::SetUp() {
+    std::vector<ov::Shape> inputShape;
+    std::pair<std::vector<ov::element::Type>, std::vector<ov::element::Type>> types;
+    std::tie(inputShape, types, ref_num_nodes, ref_num_subgraphs, targetDevice) = this->GetParam();
+    init_input_shapes(static_shapes_to_test_representation(inputShape));
+
+    auto f = ov::test::snippets::ConvertInputFunction(inputShape, types.first[0], types.second[0]);
+    function = f.getOriginal();
+}
+
+parameters ConvertInput::generate_params_random() const {
+    parameters params;
+    const auto& funcInputs = function->inputs();
+    for (int i = 0; i < funcInputs.size(); ++i) {
+        int32_t startFrom, range, resolution = 1;
+        switch (funcInputs[i].get_element_type()) {
+            case ov::element::f32:
+            case ov::element::bf16:
+                startFrom = -10;
+                range = 20;
+                resolution = 7;
+                break;
+            case ov::element::i32:
+            case ov::element::i8:
```

Comment:
You added this Saturation/Truncation algorithms to fix accuracy issues. Do we really check somewhere that corner cases (like 257 value conversions) works according to ngraph spec? I think generic CPU Convert single layer test is a good place for that.

Reply:
Sure. For that we have the separate test with only one Convert and with the [separate](https://github.com/a-sidorova/openvino/blob/1bf040307c4394a03d0e04021cb67956c8c5d8bf/src/tests/functional/plugin/shared/src/snippets/convert.cpp#L54) `generate_params_random()`. So I verify correct calculations of negative values in U8 case and  of values are closer to 128 for I8 case